keyvi/tests/cpp/stringdistance/distance_matrix_test.cpp

```diff
@@ -0,0 +1,48 @@
+//
+// keyvi - A key value store.
+//
+// Copyright 2015 Hendrik Muhs<hendrik.muhs@gmail.com>
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+/*
+ * distance_matrix_test.cpp
+ *
+ *  Created on: Jan 17, 2017
+ *      Author: konstantin.s
+ */
+
+#include <boost/test/unit_test.hpp>
+#include "stringdistance/distance_matrix.h"
+
+namespace keyvi {
+namespace stringdistance {
+
+BOOST_AUTO_TEST_SUITE( DistanceMatrixTests )
+
+void TestResizeOverflow(DistanceMatrix& matrix) {
+  const size_t new_rows_number = 1LL + (1LL << 32LL);
+  matrix.EnsureRowCapacity(new_rows_number);
```

Comment:
The valid point you brought up here - intentional or not - : should we harden the APIs and e.g. put limits into the API to e.g. disallow key length > X? I actually do not think we are at a point that we need that. I believe it is common sense to not do 'stupid stuff' like this. 

It would make sense to do this in e.g. "keyvi-server" as this a potential security threat. I will keep it in mind for it.